internal/controller/servermaintenance_controller_test.go

```diff
@@ -0,0 +1,207 @@
+					metalv1alpha1.ServerMaintenanceReasonAnnotationKey: "test-maintenance",
+				},
+			},
+			Spec: metalv1alpha1.ServerMaintenanceSpec{
+				ServerRef:   &v1.LocalObjectReference{Name: server.Name},
+				Policy:      metalv1alpha1.ServerMaintenancePolicyEnforced,
+				ServerPower: metalv1alpha1.PowerOff,
+				ServerBootConfigurationTemplate: &metalv1alpha1.ServerBootConfigurationTemplate{
+					Name: "test-boot",
+					Spec: metalv1alpha1.ServerBootConfigurationSpec{
+						ServerRef: v1.LocalObjectReference{Name: server.Name},
+						Image:     "some_image",
+					},
+				},
+			},
+		}
+		Expect(k8sClient.Create(ctx, serverMaintenance)).To(Succeed())
+		Eventually(Object(serverMaintenance)).Should(SatisfyAll(
+			HaveField("Status.State", metalv1alpha1.ServerMaintenanceStateInMaintenance),
+		))
+
+		By("Checking the Server is in maintenance")
+		Eventually(Object(server)).Should(SatisfyAll(
+			HaveField("Status.State", metalv1alpha1.ServerStateMaintenance),
+		))
+	})
+
+	It("Should wait to put a Server into maintenance until approval", func(ctx SpecContext) {
+		By("Creating an ServerMaintenance object")
+		serverMaintenance := &metalv1alpha1.ServerMaintenance{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-server-maintenance",
+				Namespace: ns.Name,
+				Annotations: map[string]string{
+					metalv1alpha1.ServerMaintenanceReasonAnnotationKey: "test-maintenance",
+				},
+			},
+			Spec: metalv1alpha1.ServerMaintenanceSpec{
+				ServerRef:   &v1.LocalObjectReference{Name: server.Name},
+				Policy:      metalv1alpha1.ServerMaintenancePolicyOwnerApproval,
+				ServerPower: metalv1alpha1.PowerOff,
+				ServerBootConfigurationTemplate: &metalv1alpha1.ServerBootConfigurationTemplate{
+					Name: "test-boot",
+					Spec: metalv1alpha1.ServerBootConfigurationSpec{
+						ServerRef: v1.LocalObjectReference{Name: server.Name},
+						Image:     "some_image",
+					},
+				},
+			},
+		}
+		serverClaim := &metalv1alpha1.ServerClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-claim",
+				Namespace: ns.Name,
+			},
+		}
+		Expect(k8sClient.Create(ctx, serverClaim)).To(Succeed())
+		By("Patching the Server to reserved state")
+		server.Spec.ServerClaimRef = &v1.ObjectReference{
+			Name:      serverClaim.Name,
+			Namespace: serverClaim.Namespace,
+		}
+		Expect(k8sClient.Update(ctx, server)).To(Succeed())
+		Eventually(UpdateStatus(server, func() {
+			server.Status.State = metalv1alpha1.ServerStateReserved
+		})).Should(Succeed())
+		Expect(k8sClient.Create(ctx, serverMaintenance)).To(Succeed())
+		Eventually(Object(serverMaintenance)).Should(SatisfyAll(
+			HaveField("Status.State", metalv1alpha1.ServerMaintenanceStatePending),
+		))
+
+		By("Checking the Server is not in maintenance")
+		Eventually(Object(server)).Should(SatisfyAll(
+			HaveField("Status.State", metalv1alpha1.ServerStateReserved),
+		))
+
+		By("Approving the maintenance")
+		Expect(k8sClient.Get(ctx, types.NamespacedName{
```

Comment:
I don't think we need to do a `Get` here. Instead tely on the `Update` helper to just update the annotation. If the object is not present it will throw an error.